rueidisprob/countingbloomfilter.go

```diff
@@ -0,0 +1,375 @@
+	if resp.Error() != nil {
+		return resp.Error()
+	}
+
+	return nil
+}
+
+func (f *countingBloomFilter) indexes(keys []string) []string {
+	allIndexes := make([]string, 0, len(keys)*int(f.hashIterations))
+	size := uint64(f.size)
+	for _, key := range keys {
+		h1, h2 := hash([]byte(key))
+		for i := uint(0); i < f.hashIterations; i++ {
+			allIndexes = append(allIndexes, strconv.FormatUint(index(h1, h2, i, size), 10))
+		}
+	}
+	return allIndexes
+}
+
+func (f *countingBloomFilter) Exists(ctx context.Context, key string) (bool, error) {
+	exists, err := f.ExistsMulti(ctx, []string{key})
+	if err != nil {
+		return false, err
+	}
+
+	return exists[0], nil
+}
+
+func (f *countingBloomFilter) ExistsMulti(ctx context.Context, keys []string) ([]bool, error) {
+	if len(keys) == 0 {
+		return nil, nil
+	}
+
+	deduplicatedKeys := make([]string, 0, len(keys))
+	keySet := make(map[string]struct{}, len(keys))
+	for _, key := range keys {
+		if _, ok := keySet[key]; !ok {
+			keySet[key] = struct{}{}
+			deduplicatedKeys = append(deduplicatedKeys, key)
+		}
+	}
+
+	indexes := f.indexes(keys)
+
+	args := make([]string, 0, len(indexes)+1)
+	args = append(args, f.hashIterationString)
+	args = append(args, indexes...)
+
+	resp := f.existsMultiScript.Exec(ctx, f.client, f.existsMultiKeys, args)
+	if resp.Error() != nil {
+		return nil, resp.Error()
+	}
+
+	arr, err := resp.AsBoolSlice()
+	if err != nil {
+		return nil, err
+	}
+	return arr, nil
+}
+
+func (f *countingBloomFilter) Remove(ctx context.Context, key string) error {
+	return f.RemoveMulti(ctx, []string{key})
+}
+
+func (f *countingBloomFilter) RemoveMulti(ctx context.Context, keys []string) error {
+	if len(keys) == 0 {
+		return nil
+	}
+
+	deduplicatedKeys := make([]string, 0, len(keys))
+	keySet := make(map[string]struct{}, len(keys))
+	for _, key := range keys {
+		if _, ok := keySet[key]; !ok {
+			keySet[key] = struct{}{}
+			deduplicatedKeys = append(deduplicatedKeys, key)
+		}
+	}
+
+	indexes := f.indexes(deduplicatedKeys)
+	args := make([]string, 0, len(indexes)+1)
+	args = append(args, f.hashIterationString)
+	args = append(args, indexes...)
+
+	resp := f.removeMultiScript.Exec(ctx, f.client, f.removeMultiKeys, args)
+	if resp.Error() != nil {
+		return resp.Error()
+	}
+
+	return nil
+}
+
+func (f *countingBloomFilter) Delete(ctx context.Context) error {
+	resp := f.client.Do(
+		ctx,
+		f.client.B().
+			Eval().
+			Script(countingBloomFilterDeleteScript).
+			Numkeys(2).
+			Key(f.name, f.counter).
+			Build(),
+	)
+	if resp.Error() != nil {
+		return resp.Error()
+	}
+
+	return nil
+}
+
+func (f *countingBloomFilter) Count(ctx context.Context) (uint, error) {
+	resp := f.client.Do(
+		ctx,
+		f.client.B().
+			Get().
+			Key(f.counter).
+			Build(),
+	)
+	if resp.Error() != nil {
+		if rueidis.IsRedisNil(resp.Error()) {
+			return 0, nil
+		}
+
+		return 0, resp.Error()
+	}
+
+	count, err := resp.AsUint64()
+	if err != nil {
+		return 0, err
+	}
```

Comment:
```suggestion
	count, err := resp.AsUint64()
	if err != nil {
		if rueidis.IsRedisNil(err) {
			return 0, nil
		}
		return 0, err
	}
```
There is no need to use `resp.Error()` separately. The  `resp.AsUint64()` should also return the same err.

Reply:
[7069d90](https://github.com/redis/rueidis/pull/519/commits/7069d905ebc9d7bcc0eed498d6755ef8735d752f)